core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -246,6 +249,15 @@ class HadoopRDD[K, V](
         } catch {
           case eof: EOFException =>
             finished = true
+          case e: Exception =>
```

Comment:
Having been on the receiving end of things I know that the gzip module throws an IOException, but unfortunately I have no knowledge over what the Hadoop input modules and what exceptions they throw, or if they propagate exceptions up from other 3rd party libraries.   Catching such a broad exception is mitigated by the fact that this particular option defaults to off, and should only be enabled when you are trying to parse files that you know are corrupt.  Given the situation, when you turn the option on, we should really try to finish processing files to the best of our ability, thus I think in this case catching 'Exception' might be appropriate.